### PR TITLE
Resolution fails on Windows for interfaces extending another typed interface.

### PR DESCRIPTION
**Pull request by maartenc**
Constructs like 

`public interface A extends B<C> {}`

results in 

`java.nio.file.InvalidPathException: Illegal char <<>`

exceptions on windows.

### CODE REVIEW

**Comment by maartenc** [comment]
It looks like this issue has already been reported here: #1948 and #1876

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserInterfaceDeclaration.java [hunk 1]
```diff
@@ -353,14 +353,18 @@ public ClassOrInterfaceDeclaration getWrappedNode() {
 
     private ResolvedReferenceType toReferenceType(ClassOrInterfaceType classOrInterfaceType) {
         SymbolReference<? extends ResolvedTypeDeclaration> ref = null;
-        String typeName = classOrInterfaceType.asString();
+        String typeName = classOrInterfaceType.getName().getId();
+        if (classOrInterfaceType.getScope().isPresent()) {
+            typeName = classOrInterfaceType.getScope().get().toString() + "." + typeName;
```
Comment (matozoid) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserInterfaceDeclaration.java]: Please use asString here, see http://javaparser.org/making-strings-in-javaparser/

File: javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.function.BiConsumer;
+
+/**
+ * A validator that validates a known node type.
+ */
+public interface TypedValidator<N extends Node> extends BiConsumer<N, ProblemReporter> {
+    /**
+     * @param node the node that wants to be validated
+     * @param problemReporter when found, validation errors can be reported here
```
Comment (matozoid) [inline javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java]: @ftomassetti is it okay to add new files in this directory?

**Comment by matozoid** [comment]
Thanks!

**Comment by maartenc** [comment]
ok, I've changed the `toString()` to `asString()`.

I did copy/past this code from the `JavaParserClassDeclaration#toReferenceType()` method, perhaps you should change it there as well?

Maarten

**Comment by matozoid** [comment]
Okay, made a note to check for toString()s. This is now waiting for @ftomassetti (who seems incredibly busy)

File: javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java (hunk 2 above)
Comment (ftomassetti) [reply javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java]: No, it should basically be inserted anywhere else but here :)

Comment (ftomassetti) [reply javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java]: Of course also the Validator.java file should be moved. I would create a directory under the root of the resources folder and put both files there

**Review by ftomassetti — COMMENTED** [review]
+1 if we move the files

**Comment by maartenc** [comment]
Ok, I'll move the files and simplify them. 
I wanted to make this test similar to the other tests in that file, but I didn't knew that it was not ok to add new files to that directory.

I'll update this PR when I'm finished.

thanks for the feedback,
Maarten

**Comment by maartenc** [comment]
I've removed the added files from the javaparser_new_src directory and reused some others I found under the resources directory. I hope that is ok?

**Comment by matozoid** [comment]
+1 from me then!

**Comment by maartenc** [comment]
Thanks Danny, maybe you could close the mentioned issues as well?
(#1948 and #1876)

Maarten

**Pull request merged**